tests/unit/scripting.tcl

```diff
@@ -1404,71 +1404,117 @@ start_server {tags {"scripting"}} {
 # Additional eval only tests
 start_server {tags {"scripting"}} {
     test "Consistent eval error reporting" {
+        r config resetstat
         r config set maxmemory 1
         # Script aborted due to Redis state (OOM) should report script execution error with detailed internal error
-        assert_error {ERR Error running script (call to *): @user_script:*: OOM command not allowed when used memory > 'maxmemory'.} {
+        assert_error {OOM command not allowed when used memory > 'maxmemory'*} {
             r eval {return redis.call('set','x','y')} 1 x
         }
+        assert_equal [errorrstat OOM r] {count=1}
+
         # redis.pcall() failure due to Redis state (OOM) returns lua error table with Redis error message without '-' prefix
+        r config resetstat
         assert_equal [
             r eval {
                 local t = redis.pcall('set','x','y')
-                if t['err'] == "OOM command not allowed when used memory > 'maxmemory'." then
+                if t['err'] == "command not allowed when used memory > 'maxmemory'." then
```

Comment:
same here.. do we want better backwards compatibility? can we?

Reply:
We can, but I think its a good time to "break" it on Redis 7, its only the error message content so I do not think its a big change and I believe its more accurate (remember the user can still get the error code on `t['err_code']`).

Reply:
our intention was to make sure the error code always exists, and that error stats are not counted twice.
along the way we created a map (mainly for the stat), and added a separate field for the error code, but i don't think it means that we meant to trim the error code from the message, and cause breaking changes in places that were not necessary.